<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>cmm studio</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QSplitter" name="splitter_3">
      <property name="lineWidth">
       <number>0</number>
      </property>
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="handleWidth">
       <number>0</number>
      </property>
      <widget class="QTabWidget" name="tabWidget">
       <property name="currentIndex">
        <number>0</number>
       </property>
       <widget class="QWidget" name="projectWidget">
        <attribute name="title">
         <string>项目</string>
        </attribute>
        <layout class="QGridLayout" name="gridLayout_3">
         <property name="leftMargin">
          <number>0</number>
         </property>
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="rightMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="ProjectTreeView" name="projectView">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
             <horstretch>0</horstretch>
             <verstretch>100</verstretch>
            </sizepolicy>
           </property>
           <property name="contextMenuPolicy">
            <enum>Qt::CustomContextMenu</enum>
           </property>
           <property name="editTriggers">
            <set>QAbstractItemView::EditKeyPressed</set>
           </property>
           <property name="headerHidden">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
      <widget class="QSplitter" name="splitter_2">
       <property name="lineWidth">
        <number>0</number>
       </property>
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="handleWidth">
        <number>0</number>
       </property>
       <widget class="QSplitter" name="splitter">
        <property name="lineWidth">
         <number>0</number>
        </property>
        <property name="orientation">
         <enum>Qt::Vertical</enum>
        </property>
        <property name="handleWidth">
         <number>0</number>
        </property>
        <widget class="QTabWidget" name="docTabWidget">
         <property name="currentIndex">
          <number>0</number>
         </property>
         <property name="tabsClosable">
          <bool>true</bool>
         </property>
         <property name="movable">
          <bool>true</bool>
         </property>
         <widget class="QWidget" name="tab_3">
          <attribute name="title">
           <string>no documents</string>
          </attribute>
         </widget>
        </widget>
        <widget class="QTabWidget" name="resultWidget">
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="currentIndex">
          <number>0</number>
         </property>
         <widget class="QWidget" name="outputWidget">
          <attribute name="title">
           <string>输出</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_6">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QTextBrowser" name="outputBrowser">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
               <horstretch>0</horstretch>
               <verstretch>1</verstretch>
              </sizepolicy>
             </property>
             <property name="font">
              <font>
               <pointsize>12</pointsize>
              </font>
             </property>
             <property name="lineWrapMode">
              <enum>QTextEdit::NoWrap</enum>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="stackWidget">
          <attribute name="title">
           <string>栈</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_7">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QTableView" name="stackView">
             <property name="selectionMode">
              <enum>QAbstractItemView::NoSelection</enum>
             </property>
             <attribute name="verticalHeaderVisible">
              <bool>false</bool>
             </attribute>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="globalWidget">
          <attribute name="title">
           <string>全局区</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_8">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QTableView" name="globalView">
             <property name="selectionMode">
              <enum>QAbstractItemView::NoSelection</enum>
             </property>
             <attribute name="verticalHeaderVisible">
              <bool>false</bool>
             </attribute>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="constWidget">
          <attribute name="title">
           <string>常量区</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_9">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QTableView" name="constView">
             <property name="selectionMode">
              <enum>QAbstractItemView::NoSelection</enum>
             </property>
             <property name="sortingEnabled">
              <bool>false</bool>
             </property>
             <attribute name="verticalHeaderVisible">
              <bool>false</bool>
             </attribute>
            </widget>
           </item>
          </layout>
         </widget>
         <widget class="QWidget" name="codeWidget">
          <attribute name="title">
           <string>代码段</string>
          </attribute>
          <layout class="QGridLayout" name="gridLayout_10">
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item row="0" column="0">
            <widget class="QTableView" name="codeView">
             <property name="selectionMode">
              <enum>QAbstractItemView::NoSelection</enum>
             </property>
             <attribute name="verticalHeaderVisible">
              <bool>false</bool>
             </attribute>
            </widget>
           </item>
          </layout>
         </widget>
        </widget>
       </widget>
       <widget class="QTabWidget" name="analysisWidget">
        <property name="font">
         <font>
          <pointsize>11</pointsize>
         </font>
        </property>
        <property name="currentIndex">
         <number>0</number>
        </property>
        <widget class="QWidget" name="lexerWidget">
         <attribute name="title">
          <string>词法分析</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_2">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QTextBrowser" name="lexerBrowser">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
              <horstretch>1</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="font">
             <font>
              <pointsize>11</pointsize>
             </font>
            </property>
            <property name="lineWrapMode">
             <enum>QTextEdit::NoWrap</enum>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="parserWidget">
         <attribute name="title">
          <string>语法分析</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_4">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <property name="horizontalSpacing">
           <number>6</number>
          </property>
          <item row="0" column="0">
           <widget class="QTreeView" name="syntacticTreeView">
            <property name="headerHidden">
             <bool>true</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
        <widget class="QWidget" name="codeGeneWidget">
         <attribute name="title">
          <string>中间代码</string>
         </attribute>
         <layout class="QGridLayout" name="gridLayout_5">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>0</number>
          </property>
          <item row="0" column="0">
           <widget class="QTextBrowser" name="codeGeneBrowser"/>
          </item>
         </layout>
        </widget>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>900</width>
     <height>23</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu_F">
    <property name="tearOffEnabled">
     <bool>false</bool>
    </property>
    <property name="title">
     <string>文件(&amp;F)</string>
    </property>
    <property name="separatorsCollapsible">
     <bool>false</bool>
    </property>
    <property name="toolTipsVisible">
     <bool>false</bool>
    </property>
    <addaction name="action_new"/>
    <addaction name="action_open"/>
    <addaction name="separator"/>
    <addaction name="action_save"/>
    <addaction name="action_saveAs"/>
    <addaction name="action_exit"/>
   </widget>
   <widget class="QMenu" name="menu_R">
    <property name="title">
     <string>运行(R)</string>
    </property>
    <addaction name="action_lexicalAnalysis"/>
    <addaction name="action_syntacticAnalysis"/>
    <addaction name="action_codeGene"/>
    <addaction name="action_run"/>
   </widget>
   <widget class="QMenu" name="menu_E">
    <property name="title">
     <string>编辑(&amp;E)</string>
    </property>
    <addaction name="action_cut"/>
    <addaction name="action_copy"/>
    <addaction name="action_paste"/>
   </widget>
   <widget class="QMenu" name="menu_D">
    <property name="title">
     <string>调试(&amp;D)</string>
    </property>
    <addaction name="action_debug"/>
    <addaction name="action_singleProcess"/>
    <addaction name="action_singleStmt"/>
    <addaction name="action_stopDebug"/>
   </widget>
   <addaction name="menu_F"/>
   <addaction name="menu_E"/>
   <addaction name="menu_D"/>
   <addaction name="menu_R"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="action_new"/>
   <addaction name="action_open"/>
   <addaction name="separator"/>
   <addaction name="action_save"/>
   <addaction name="action_saveAs"/>
   <addaction name="separator"/>
   <addaction name="action_cut"/>
   <addaction name="action_copy"/>
   <addaction name="action_paste"/>
   <addaction name="separator"/>
   <addaction name="action_run"/>
   <addaction name="action_debug"/>
   <addaction name="action_stopDebug"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="action_new">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/resource/images/new.png</normaloff>:/resource/images/new.png</iconset>
   </property>
   <property name="text">
    <string>新建文件(&amp;N)</string>
   </property>
   <property name="toolTip">
    <string>新建</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="action_lexicalAnalysis">
   <property name="text">
    <string>词法分析(&amp;L)</string>
   </property>
   <property name="toolTip">
    <string>词法分析</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+L</string>
   </property>
  </action>
  <action name="action_syntacticAnalysis">
   <property name="text">
    <string>语法分析(&amp;P)</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="action_open">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/resource/images/open.png</normaloff>:/resource/images/open.png</iconset>
   </property>
   <property name="text">
    <string>打开文件(&amp;O)</string>
   </property>
   <property name="toolTip">
    <string>打开</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="action_codeGene">
   <property name="text">
    <string>中间代码生成(&amp;G)</string>
   </property>
   <property name="toolTip">
    <string>中间代码生成</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+G</string>
   </property>
  </action>
  <action name="action_run">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/resource/images/run.png</normaloff>:/resource/images/run.png</iconset>
   </property>
   <property name="text">
    <string>运行(&amp;R)</string>
   </property>
   <property name="toolTip">
    <string>运行</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
  </action>
  <action name="action_save">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/resource/images/save.png</normaloff>:/resource/images/save.png</iconset>
   </property>
   <property name="text">
    <string>保存(&amp;S)</string>
   </property>
   <property name="toolTip">
    <string>保存</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="action_saveAs">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/resource/images/saveAs.png</normaloff>:/resource/images/saveAs.png</iconset>
   </property>
   <property name="text">
    <string>另存为(&amp;A)</string>
   </property>
   <property name="toolTip">
    <string>另存为</string>
   </property>
  </action>
  <action name="action_debug">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/resource/images/debug.png</normaloff>:/resource/images/debug.png</iconset>
   </property>
   <property name="text">
    <string>开始调试</string>
   </property>
   <property name="shortcut">
    <string>F5</string>
   </property>
  </action>
  <action name="action_singleProcess">
   <property name="text">
    <string>逐过程</string>
   </property>
   <property name="shortcut">
    <string>F10</string>
   </property>
  </action>
  <action name="action_singleStmt">
   <property name="text">
    <string>逐语句</string>
   </property>
   <property name="shortcut">
    <string>F11</string>
   </property>
  </action>
  <action name="action_stopDebug">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/resource/images/stop_debug.png</normaloff>:/resource/images/stop_debug.png</iconset>
   </property>
   <property name="text">
    <string>停止调试</string>
   </property>
  </action>
  <action name="action_exit">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/resource/images/exit.png</normaloff>:/resource/images/exit.png</iconset>
   </property>
   <property name="text">
    <string>退出(&amp;X)</string>
   </property>
  </action>
  <action name="action_cut">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/resource/images/cut.png</normaloff>:/resource/images/cut.png</iconset>
   </property>
   <property name="text">
    <string>剪切(&amp;T)</string>
   </property>
   <property name="toolTip">
    <string>剪切</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+X</string>
   </property>
  </action>
  <action name="action_copy">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/resource/images/copy.png</normaloff>:/resource/images/copy.png</iconset>
   </property>
   <property name="text">
    <string>复制(&amp;C)</string>
   </property>
   <property name="toolTip">
    <string>复制</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="action_paste">
   <property name="icon">
    <iconset resource="resource.qrc">
     <normaloff>:/resource/images/paste.png</normaloff>:/resource/images/paste.png</iconset>
   </property>
   <property name="text">
    <string>粘贴(&amp;P)</string>
   </property>
   <property name="toolTip">
    <string>粘贴</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+V</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>ProjectTreeView</class>
   <extends>QTreeView</extends>
   <header location="global">projecttreeview.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="resource.qrc"/>
 </resources>
 <connections/>
</ui>
